src/state_calc.py

```
@@ -0,0 +1,849 @@
+    def __get_cyclin_index(self, cyclin: str) -> int:
+        return self.__all_cyclins.index(cyclin)
+
+    def get_optimal_scores(self) -> tuple[int, int]:
+        return self.__optimal_graph_score, self.__optimal_g1_graph_score
+
+    def __get_all_g1_states(self) -> list:
+        return self.filter_start_states(
+            zero_cyclins=self.__g1_state_zero_cyclins, one_cyclins=self.__g1_state_one_cyclins
+        )
+
+    def filter_start_states(self, zero_cyclins: list = list(), one_cyclins: list = list()):
```

Ruff (0.11.9)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)